front/components/trackers/TrackerBuilder.tsx

```diff
@@ -0,0 +1,376 @@
+import {
+  Button,
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+  Input,
+  Label,
+  Page,
+  TextArea,
+  useSendNotification,
+} from "@dust-tt/sparkle";
+import type {
+  DataSourceViewType,
+  SpaceType,
+  SubscriptionType,
+  SupportedModel,
+  TrackerConfigurationStateType,
+  TrackerConfigurationType,
+  WorkspaceType,
+} from "@dust-tt/types";
+import {
+  CLAUDE_3_5_SONNET_DEFAULT_MODEL_CONFIG,
+  TRACKER_FREQUENCY_TYPES,
+} from "@dust-tt/types";
+import { useRouter } from "next/router";
+import { useState } from "react";
+
+import { AdvancedSettings } from "@app/components/assistant_builder/InstructionScreen";
+import AppLayout from "@app/components/sparkle/AppLayout";
+import { AppLayoutSimpleSaveCancelTitle } from "@app/components/sparkle/AppLayoutTitle";
+import { isEmailValid } from "@app/lib/utils";
+
+export const TrackerBuilder = ({
+  owner,
+  subscription,
+  globalSpace,
+  dataSourceViews,
+  trackerToEdit,
+}: {
+  owner: WorkspaceType;
+  subscription: SubscriptionType;
+  globalSpace: SpaceType;
+  dataSourceViews: DataSourceViewType[];
+  trackerToEdit: TrackerConfigurationType | null;
+}) => {
+  const router = useRouter();
+  const sendNotification = useSendNotification();
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const [tracker, setTracker] = useState<TrackerConfigurationStateType>({
+    name: trackerToEdit?.name ?? null,
+    nameError: null,
+    description: trackerToEdit?.description ?? null,
+    descriptionError: null,
+    prompt: trackerToEdit?.prompt ?? null,
+    promptError: null,
+    frequency: trackerToEdit?.frequency ?? "daily",
+    frequencyError: null,
+    recipients: trackerToEdit?.recipients?.join(", ") ?? null,
+    recipientsError: null,
+    modelId:
+      trackerToEdit?.modelId ?? CLAUDE_3_5_SONNET_DEFAULT_MODEL_CONFIG.modelId,
+    providerId:
+      trackerToEdit?.providerId ??
+      CLAUDE_3_5_SONNET_DEFAULT_MODEL_CONFIG.providerId,
+    temperature: trackerToEdit?.temperature ?? 0.5,
+  });
+
+  void dataSourceViews; // todo: use this
+
+  const extractEmails = (text: string) =>
+    text
+      .split(/[\n,]+/)
+      .map((e) => e.trim())
+      .filter((e, i, self) => self.indexOf(e) === i);
```

Comment:
nit: `[...new Set(thing)]` or `_.uniq(thing)` is a lot faster and more readable IMHO